src/gs-category-page.c

```diff
@@ -332,25 +332,31 @@ gs_category_page_create_filter_list (GsCategoryPage *self,
 	for (i = 0; i < children->len; i++) {
 		s = GS_CATEGORY (g_ptr_array_index (children, i));
 		if (gs_category_get_size (s) < 1) {
-			gboolean category_is_usb = FALSE;
+			gboolean show_no_apps_box = FALSE;
+			GPtrArray *usb_apps = NULL;
 
-			g_debug ("not showing %s/%s as no apps",
-				 gs_category_get_id (category),
-				 gs_category_get_id (s));
-
-			/* re-filter USB category with no apps so the
+ 			/* re-filter USB category with no apps so the
 			 * placeholder app tiles get cleared out, then set
 			 * "empty state" message for an empty USB disk
 			 */
 			if (g_strcmp0 (gs_category_get_id (category), "usb") == 0) {
+				g_debug ("not showing %s/%s CATEGORY USB CARAI",
+					 gs_category_get_id (category),
+					 gs_category_get_id (s));
 				gs_category_page_populate_filtered (self, s);
-				category_is_usb = TRUE;
+				usb_apps = gs_category_get_children (category);
+				show_no_apps_box = TRUE;
+			}
+			/* Then set "empty state" message for an empty USB disk. */
+			if ((usb_apps == NULL) || (usb_apps->len < 1)) {
```

Comment:
Use `usb_apps->len == 0` instead of `usb_apps->len < 1`. it is unsigned, so being `< 1` always means it's 0.